pkg/services/ngalert/schedule/compat.go

```diff
@@ -1,31 +1,53 @@
 package schedule
 
 import (
+	"fmt"
+	"net/url"
+	"path"
 	"time"
 
 	"github.com/go-openapi/strfmt"
 	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
 	"github.com/prometheus/alertmanager/api/v2/models"
 
+	"github.com/grafana/grafana/pkg/infra/log"
+	ngModels "github.com/grafana/grafana/pkg/services/ngalert/models"
 	"github.com/grafana/grafana/pkg/services/ngalert/state"
 )
 
-func FromAlertStateToPostableAlerts(firingStates []*state.State, stateManager *state.Manager) apimodels.PostableAlerts {
+func FromAlertStateToPostableAlerts(logger log.Logger, firingStates []*state.State, stateManager *state.Manager, appURL string) apimodels.PostableAlerts {
 	alerts := apimodels.PostableAlerts{PostableAlerts: make([]models.PostableAlert, 0, len(firingStates))}
 	var sentAlerts []*state.State
 	ts := time.Now()
+
+	u, err := url.Parse(appURL)
+	if err != nil {
+		logger.Debug("failed to parse URL while joining URL", "url", appURL, "err", err.Error())
+		u = nil
+	}
```

Comment:
This is also checked [here](https://github.com/grafana/grafana/blob/5e46bbe82310d92dd77865561b67e7e2c7ca8b03/pkg/setting/setting.go#L433) but it's no harm to keep it also here.

Reply:
Will keep this check here anyway to not assume anything about how data is passed to ngalert